src/Microsoft.OData.Core/PropertyAndAnnotationCollector.cs

```diff
@@ -353,7 +351,7 @@ internal void AddCustomPropertyAnnotation(string propertyName, string annotation
         {
             Debug.Assert(!string.IsNullOrEmpty(propertyName));
             Debug.Assert(!string.IsNullOrEmpty(annotationName));
-            Debug.Assert(!JsonLight.ODataJsonLightReaderUtils.IsODataAnnotationName(annotationName));
+            Debug.Assert(!Json.ODataJsonReaderUtils.IsODataAnnotationName(annotationName));

```

Comment:
```suggestion
            Debug.Assert(!ODataJsonReaderUtils.IsODataAnnotationName(annotationName));
```
You already have a `Microsoft.OData.Json` using.